spring-kafka/src/main/java/org/springframework/kafka/listener/ListenerUtils.java

```diff
@@ -185,11 +187,22 @@ public static void unrecoverableBackOff(BackOff backOff, ThreadLocal<BackOffExec
 	 * @since 2.7
 	 */
 	public static void stoppableSleep(MessageListenerContainer container, long interval) throws InterruptedException {
+		stoppableSleep(container::isRunning, interval);
+	}
+
+	/**
+	 * Sleep for the desired timeout, as long as shouldSleepCondition supplies true.
+	 * @param shouldSleepCondition to.
+	 * @param interval the timeout.
+	 * @throws InterruptedException if the thread is interrupted.
+	 * @since 3.0.9
+	 */
+	public static void stoppableSleep(Supplier<Boolean> shouldSleepCondition, long interval) throws InterruptedException {
```

Comment:
Perhaps rename to `conditionalSleep()` ?

Reply:
yeah, agree